Appz/Appz/Apps/Mail.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  Mail.swift
+//  Appz
+//
+//  Created by Suraj Shirvankar on 12/5/15.
+//  Copyright © 2015 kitz. All rights reserved.
+//
+
+import Foundation
+
+
+public extension AvailableApplications {
+    
+    public struct Mail: ExternalApplication {
+        
+        public let scheme = "mailto:"
+        public let fallbackURL: String? = ""
+        
+        
+        public struct Email {
+            var to: String? = ""
+            var subject: String? = ""
+            var body: String? = ""
+            
+            init(to: String, subject: String, body: String){
+                self.to = to
+                self.subject = subject
+                self.body = body
+            }
+            init(to: String){
+                self.to = to
+            }
+            
+            init(subject: String){
+                self.subject = subject
+            }
+            
+            func toUri() -> String{
+                return String(format: "%@?subject=%@&body=%@",
+                    self.to!, self.subject!, self.body!)
```

Comment:
1. If the end developer assigns `nil` to any of these properties, then attempts to use it ... :bomb:. We can't allow that. We can make them non-optional like so:

```swift
struct Email {
    var to: String
    var subject: String
    var body: String

    init(to: String = "", subject: String = "", body: String = "") {
        self.to = to
        self.subject = subject
        self.body = body
    }
}
```

2. Minor detail, let's try not using `self` unless necessary
3. Minor detail, I appreciate that you are encapsulating logic in a "fat model", however I feel this logic is expected to be located in the `path` property, for consistency with other Appz